app/src/main/java/com/odysee/app/ui/findcontent/AllContentFragment.java

```diff
@@ -466,6 +496,172 @@ public void onPause() {
+                                @Override
+                                public void run() {
+                                    Snackbar.make(root, getResources().getString(R.string.snackbar_new_version_available), BaseTransientBottomBar.LENGTH_INDEFINITE).setAction(getResources().getString(R.string.snackbar_install_button), new View.OnClickListener() {
+                                        @Override
+                                        public void onClick(View view) {
+                                            Context c = getContext();
+                                            if (c != null) {
+                                                ContextCompat.checkSelfPermission(c, Manifest.permission.WRITE_EXTERNAL_STORAGE);
+                                                String filePath = c.getExternalFilesDir(null).getAbsolutePath().concat("/odysee-android.apk");
+                                                final Uri uri = Uri.parse("file://" + filePath);
+                                                File f = new File(filePath);
+
+                                                if (f.exists()) {
+                                                    f.delete();
+                                                }
+                                                DownloadManager.Request request = new DownloadManager.Request(Uri.parse(latestUrl));
+                                                request.setDescription("Odysee new version available");
+                                                request.setTitle(getResources().getString(R.string.downloadmanager_new_version_available));
+
+                                                //set destination
+                                                request.setDestinationUri(uri);
+                                                request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE);
+
+                                                // get download service and enqueue file
+                                                final DownloadManager manager = (DownloadManager) c.getSystemService(Context.DOWNLOAD_SERVICE);
+                                                manager.enqueue(request);
+
+                                                //set BroadcastReceiver to install app when .apk is downloaded
+                                                BroadcastReceiver onComplete = new BroadcastReceiver() {
+                                                    public void onReceive(Context ctxt, Intent intent) {
+                                                        ContextCompat.checkSelfPermission(ctxt, Manifest.permission.READ_EXTERNAL_STORAGE);
+
+                                                        installApk();
+                                                        getContext().unregisterReceiver(this);
+                                                    }
+                                                };
```

Comment:
Simpler code not using string concatenation and converting from Uri to File to Uri (including the extra line after the BroadcastReceiver to have a blank line before registering the receiver):
```suggestion
                                                ContextCompat.checkSelfPermission(c, Manifest.permission.WRITE_EXTERNAL_STORAGE);
                                                File file = new File(c.getExternalFilesDir(null), "odysee-android.apk");

                                                if (file.exists()) {
                                                    file.delete();
                                                }

                                                DownloadManager.Request request = new DownloadManager.Request(Uri.parse(latestUrl));
                                                request.setTitle(getResources().getString(R.string.downloadmanager_downloading_new_version));

                                                //set destination
                                                request.setDestinationUri(Uri.fromFile(file));
                                                request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE);

                                                // get download service and enqueue file
                                                final DownloadManager manager = (DownloadManager) c.getSystemService(Context.DOWNLOAD_SERVICE);
                                                manager.enqueue(request);

                                                //set BroadcastReceiver to install app when .apk is downloaded
                                                BroadcastReceiver onComplete = new BroadcastReceiver() {
                                                    public void onReceive(Context context, Intent intent) {
                                                        ContextCompat.checkSelfPermission(context, Manifest.permission.READ_EXTERNAL_STORAGE);

                                                        installApk(file);
                                                        getContext().unregisterReceiver(this);
                                                    }
                                                };

```

Reply:
Using a string to hold the path makes it easier to debug any problem in the future.

I will add the extra line.

Reply:
> Using a string to hold the path makes it easier to debug any problem in the future.

Why? You can still print `file.getAbsolutePath()`

Reply:
@kekkyojin ^

Reply:
Printing some string value into logcat is not debugging. When debugging, you would need to see at every moment what value is stored on any variable.

Reply:
You can still debug the `file` variable. I don't see the need to have a string that gets converted to a `File`.

In Android Studio you can enter custom expressions in the debugger, so you can watch the value of `file.getAbsolutePath()`.